slicerator.py

```diff
@@ -108,7 +172,8 @@ def __getitem__(self, key):
             rel_indices = range(start, stop, step)
             new_length = len(rel_indices)
             indices = _index_generator(rel_indices, abs_indices)
-            return Slicerator(self._ancestor, indices, new_length)
+            return Slicerator(self._ancestor, '__getitem__', indices, new_length,
```

Comment:
@danielballan This keeps bugging me: why is the first argument `self._ancestor` and not `self`?